### PR TITLE
[codediff] Gracefully handle starting from detached HEAD

### PR DESCRIPTION
**Pull request by jacobhinkle**
Currently if you are on a detached HEAD (i.e. if you have run `git checkout <SHA>` then `compare_codegen.sh` will fail with `fatal: ref HEAD is not a symbolic ref`. This PR makes it safe to run in such a scenario.

NOTE: I am sometimes seeing new git branches introduced at cleanup named like the SHA of the commit and the compare commit. This has happened previously and I think it is unrelated to this change. Since it only affects cleanup, I would like to merge this PR so that we can improve `!build --diff` in the CI and look into the branch issue later.

### CODE REVIEW

File: tools/codediff/compare_codegen.sh [hunk 1]
```diff
@@ -121,7 +122,7 @@ scriptdir=$(mktemp -d -t codediffXXXXXX)
 cp -r "$nvfuserdir/tools/codediff/"* "$scriptdir/"
 
 movecudafiles() {
-    find . -maxdepth 1 -name '__tmp_kernel*.cu' -o -name '__tmp_kernel*.ptx' -exec mv '{}' "$1" \;
+    find . -maxdepth 1 \( -name '__tmp_kernel*.cu' -o -name '__tmp_kernel*.ptx' \) -exec mv '{}' "$1" \;
```
Comment (jacobhinkle) [inline tools/codediff/compare_codegen.sh:125]: Unrelated fix for a bug found by `shellcheck`.

**Review by xwang233 — APPROVED** [review]
LGTM. Thanks for the fix.
